app/services/embedded_content_finder_service.rb

```diff
@@ -24,23 +25,44 @@ def find_content_references(value)
 
   def live_content_ids(content_references)
     found_editions = live_editions(content_references.uniq)
-    not_found_content_ids = content_references.map(&:content_id) - found_editions.map(&:content_id)
+    not_found_content_ids = content_references.map(&:identifier) - found_editions.map(&:content_id)
 
     if not_found_content_ids.any?
       GovukError.notify(CommandError.new(
                           code: 422,
                           message: "Could not find any live editions for embedded content IDs: #{not_found_content_ids.join(', ')}",
                         ))
     end
-    content_references.map(&:content_id) - not_found_content_ids
+    content_references.map(&:identifier) - not_found_content_ids
```

Comment:
minor/not really related but just trying to see any way to simplify! could we just return `found_editions.map(&:content_id)` here? and assign that earlier up to a variable so that we don't have to do the map twice?

Reply:
We need to return the identifiers rather than the found editions because we need to return duplicates if the same content block is referenced twice (see https://github.com/alphagov/publishing-api/pull/3129). That said, there's definitely scope to improve the logic, so I've added another commit. Will deploy to integration first to check it out, but it should be good 👍